validator/db/db.go

```diff
@@ -62,10 +62,21 @@ func createBuckets(tx *bolt.Tx, buckets ...[]byte) error {
 	return nil
 }
 
-// NewKVStore initializes a new boltDB key-value store at the directory
-// path specified, creates the kv-buckets based on the schema, and stores
-// an open connection db object as a property of the Store struct.
-func NewKVStore(dirPath string, pubKeys [][48]byte) (*Store, error) {
+// NewKVStoreWithPublicKeyBuckets initializes a new boltDB key-value store at the directory
+// path specified, creates the kv-buckets based on the schema and provided public keys,
+// and stores an open connection db object as a property of the Store struct.
+func NewKVStoreWithPublicKeyBuckets(dirPath string, pubKeys [][48]byte) (*Store, error) {
```

Comment:
why are we using 2 separate methods to initialize the db here ?

Reply:
You're right, this is unnecessary. I created a new method because the merge function needs an empty store, so there are no keys to provide. But I should have used the old method and pass an empty slice for the `pubKeys` parameter...

Reply:
ok thanks :+1: